example/messaging/index.html

```diff
@@ -0,0 +1,59 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Messaging demo</title>
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
+          integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
+    <style type="text/css">
+        ul {
+            list-style: none;
+            margin: 0;
+            padding: 0;
+        }
+
+        li {
+            margin: 0;
+            padding: 4px 2px;
+            border-bottom: 1px solid silver;
+            overflow: auto;
+        }
+        label {
+            width: 100%;
+        }
+        input {
+            width: 100%;
+        }
+    </style>
+</head>
+<body>
+<div class="container">
+    <h1>Messaging demo</h1>
+    <ul>
+        <li>
+            <label>
+                Token
+                <input disabled id="token" />
+            </label>
+        </li>
+        <li>
+            <label>
+                Notification permission
+                <input disabled id="permission" />
+            </label>
+        </li>
+    </ul>
+    <div>
+        <button id="new_notification">Send notification</button>
```

Comment:
Disable this button unless folks provide things above? Are both required?

Reply:
in fact the two input are informal and for testing purpose.
The permission input is only to know show if the app can send notification.
And the token one is automatically populate when you do firebase.requestPermission()
I will disable the button until the permission is granted.